### DISCUSSION TITLE
Events missing for before and after execution for activity

### DISCUSSION THREAD

**Discussion by rairafaqatali** [elsa-workflows/elsa-core#2542] [Q&A]
Hi, @sfmskywalker 
I have checked the document but I was not able to find any middleware which I can use to capture events for the activity.
I want a middleware that can be used to capture that activity is started and now activity is completed? I'm using Elsa designer

**Comment by sfmskywalker** [comment]
Hi @rairafaqatali

Elsa 2 does not have a concept of middleware, but it does publish the following domain events you can handle from your application:

- ActivityExecuting
- ActivityExecuted

Your handler might look like this:

```csharp
public class SampleHandler: INotificationHandler<ActivityExecuting>, INotificationHandler<ActivityExecuted>
{
    public Task Handle(ActivityExecuting notification, CancellationToken cancellationToken)
    {
        // Activity is about to execute.
        return Task.CompletedTask;
    }

    public Task Handle(ActivityExecuted notification, CancellationToken cancellationToken)
    {
        // Activity has executed.
        return Task.CompletedTask;
    }
}
```
